nix/shared.nix

```diff
@@ -0,0 +1,120 @@
+{ compiler ? "ghc865" }:
+
+let
+  overlayShared = pkgsNew: pkgsOld: {
+    #libtorch =
+    #  let src = pkgsOld.fetchFromGitHub {
+    #      owner  = "stites";
+    #      repo   = "pytorch-world";
+    #      rev    = "4c4bd205e47477c723209f3677bddceabb614237";
+    #      sha256 = "1zqhm0874gfs1qp3glc1i1dswlrksby1wf9dr4pclkabs0smbqxc";
+    #    };
+    #  in
+    #  (import "${src}/release.nix" { }).libtorch;
+    pytorch = pkgsOld.python3Packages.pytorchWithoutCuda.override {
```

Comment:
It looks like this is referencing the older pytorch-0.4 package, but is also [using `.dev` outputs](https://github.com/hasktorch/hasktorch/pull/165/files#diff-c156cac2b10c7abb0ec25d4ec2622875R56)  generated in the pytorch-world package. Where do these come from?

Reply:
hi @stites, I'm not sure I follow. Why do you think the older pytorch-0.4 package is referenced here?
The `nixpkgs.json` is pinning `<nixpkgs>` to https://github.com/NixOS/nixpkgs/commit/9420c33455c5b3e0876813bdd739bc75b3ccbd8a, i.e. your `move work from pytorch-world` commit from 5 days ago. This is what `pkgsOld` refers to.

Reply:
oops! okay. I didn't realize my commits would be added to the main nixos/nixpkgs repo! I thought they would reference my fork.

Just so you know, I think nixpkgs has a requirement that all PRs are squashed to a single commit (pulling down the repo is already a massive undertaking). You're probably better off tracking pytorch-world for the time being, then moving back to the merged commit. I can also do this when I rebase on top of @wavewave's commits (I will be sure to keep author commits as git commits so that it doesn't mess up the git history).